src/timeline.ts

```diff
@@ -305,17 +337,87 @@ class Timeline implements IAppView {
                   open2dHistogram = this.parentNode;
                 }
               });
+           }
 
-          });
-      });
-    }
+           //all changes
+           drawBars(data, color);
+          //helper variable for clicking remove button!
+           var btnClicked = 0;
+
+            $('.btn-removed').on('click', function (e) {
```

Comment:
Avoid copy source code, try to generalize and configure instead.
Sample code (not tested): 

```typescript
let isBtnClicked = false;

d3.selectAll('.nav-bar buttons').on('click', function() {
  const button = d3.select(this);
  const changeType = button.attr('data-change-type');

  let colors = color_all;
  let data = data_all;

  svgRatioBar.selectAll('rect').remove();

  // show default all bars and colors
  if(isBtnClicked) {
    drawBars(data, colors);
    isBtnClicked = false;
    return; // exit function
  }

  // override colors for change type
  switch (changeType) {
    case 'nochanges':
      colors = color_nochanges;
      data = data_nochange;
      break;
    case 'removed':
      //...
      break;
  }

  drawBars(data, colors);
  isBtnClicked = true;
});
```

Add `data-change-type` to buttons
```typescript
<button type="button" class="btn-nochanges" data-change-type="nochanges">No Changes</button>
...
```